util/reggen/top_uvm_reg.sv.tpl

```diff
@@ -115,7 +115,10 @@ ${make_ral_pkg_window_class(dv_base_prefix, 'chip', window)}
         esc_if_name = block.name.lower() + if_suffix
         if_inst = inst_name + if_suffix
 
-        hdl_path = 'tb.dut.top_earlgrey.u_' + inst_name
+        if inst_name in top.attrs and top.attrs[inst_name] == "reggen_only":
```

Comment:
Nit: You can do this more simply with `if top.attrs.get(inst_name) == 'reggen_only'`

Reply:
This is better. Will update. Thanks